src/bin/lttng-sessiond/kernel.c

```diff
@@ -191,6 +193,227 @@ int kernel_create_channel(struct ltt_kernel_session *session,
+end:
+	return ret;
+}
+
+/*
+ * Compute the offsets of the instrumentation bytes in the binary based on the
+ * tracepoint probe location using the SDT lookup method. This function
+ * allocates the offsets buffer, the caller must free it.
+ *
+ * Returns 0 on success and set the offset out parameter to the offsets of the
+ * SDT tracepoint.
+ * Returns -1 on error.
+ */
+static
+int extract_userspace_probe_offset_tracepoint_sdt(
+		struct lttng_userspace_probe_location *probe_location,
+		struct ltt_kernel_session *session, uint64_t **offsets,
+		uint32_t *num_offset)
+{
+	int ret, fd;
+	const char *probe_name = NULL, *provider_name = NULL;
+	struct lttng_userspace_probe_location_lookup_method *lookup = NULL;
+	enum lttng_userspace_probe_location_lookup_method_type lookup_method_type;
+
+	ret = 0;
+
+	assert(lttng_userspace_probe_location_get_type(probe_location) ==
+			LTTNG_USERSPACE_PROBE_LOCATION_TYPE_TRACEPOINT);
+
+	lookup = lttng_userspace_probe_location_get_lookup_method(probe_location);
+	if (!lookup) {
+		ret = -1;
+		goto end;
+	}
+
+	lookup_method_type =
+		lttng_userspace_probe_location_lookup_method_get_type(lookup);
+
+	assert(lookup_method_type ==
+			LTTNG_USERSPACE_PROBE_LOCATION_LOOKUP_METHOD_TYPE_TRACEPOINT_SDT);
+
+
+	probe_name = lttng_userspace_probe_location_tracepoint_get_probe_name(
+			probe_location);
+	if (!probe_name) {
+		ret = -1;
+		goto end;
+	}
+
+	provider_name = lttng_userspace_probe_location_tracepoint_get_provider_name(
+			probe_location);
+	if (!provider_name) {
+		ret = -1;
+		goto end;
+	}
+
+	fd = lttng_userspace_probe_location_tracepoint_get_binary_fd(probe_location);
+	if (fd < 0) {
+		ret = -1;
+		goto end;
+	}
+
+	ret = run_as_extract_sdt_probe_offsets(fd, provider_name, probe_name,
+			session->uid, session->gid, offsets, num_offset);
+	if (ret < 0) {
+		DBG("userspace probe offset calculation failed for sdt "
+			"probe %s:%s", provider_name, probe_name);
+		goto end;
+	}
+
+	for (int i = 0; i < *num_offset; i++) {
+		DBG("userspace probe sdt offsets found for %s:%s at 0x%lx.",
+				provider_name, probe_name, (*offsets)[i]);
+	}
+end:
+	return ret;
+}
+
+/*
+ * Extract the offsets of the instrumentation point for the different lookup
+ * methods.
+ */
+static
+int userspace_probe_add_callsites(struct lttng_event *ev,
+			struct ltt_kernel_session *session, int fd)
+{
+	int ret;
+
+	struct lttng_userspace_probe_location *location = NULL;
+	struct lttng_userspace_probe_location_lookup_method *lookup_method = NULL;
+	enum lttng_userspace_probe_location_lookup_method_type type;
+
+	assert(ev);
+	assert(ev->type == LTTNG_EVENT_USERSPACE_PROBE);
+
+	location = lttng_event_get_userspace_probe_location(ev);
+	if (!location) {
+		ret = -1;
+		goto end;
+	}
+	lookup_method =
+		lttng_userspace_probe_location_get_lookup_method(location);
+	if (!lookup_method) {
+		ret = -1;
+		goto end;
+	}
+	type = lttng_userspace_probe_location_lookup_method_get_type(lookup_method);
```

Comment:
Put an empty line before `type` and remove the one after.

Reply:
done